backend/libraries/ballerina-core/codegen/python/language constructs/union.fs

```diff
@@ -0,0 +1,57 @@
+namespace Ballerina.DSL.Codegen.Python.LanguageConstructs
+
+
+module Union =
+  open Ballerina.Core.StringBuilder
+  open Ballerina.Collections.NonEmptyList
+  open Ballerina.DSL.Codegen.Python.LanguageConstructs.Model
+
+  let private indent = (+) "    "
+
+  let private appendCaseName (allCases: string) (nextCase: string) = $"{allCases} | {nextCase}"
+
+  type PythonUnion =
+    { Name: string
+      Cases: NonEmptyList<{| Name: string; Type: string |}> }
```

Comment:
nothing ensures uniqueness here

Reply:
These are already type checked, also the resulting code still needs to type check

Reply:
far